pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -106,36 +98,45 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 
 			newRule := apimodels.Rule{
 				Name:           rule.Title,
-				Labels:         nil,  // TODO: NG AlertRule does not have labels but does have annotations
+				Labels:         rule.Labels,
 				Health:         "ok", // TODO: update this in the future when error and noData states are being evaluated and set
 				Type:           apiv1.RuleTypeAlerting,
-				LastEvaluation: time.Time{}, // TODO: set this to be rule evaluation time once it is being set
-				EvaluationTime: 0,           // TODO: set this once we are saving it or adding it to evaluation results
+				LastEvaluation: time.Time{},
 			}
-			for _, instance := range instanceQuery.Result {
-				activeAt := instance.CurrentStateSince
+
+			for _, alertState := range stateMap[rule.UID] {
+				activeAt := alertState.StartsAt
 				alert := &apimodels.Alert{
-					Labels:      map[string]string(instance.Labels),
-					Annotations: nil, // TODO: set these once they are added to evaluation results
-					State:       translateInstanceState(instance.CurrentState),
+					Labels:      map[string]string(alertState.Labels),
+					Annotations: alertState.Annotations,
+					State:       alertState.State.String(),
 					ActiveAt:    &activeAt,
 					Value:       "", // TODO: set this once it is added to the evaluation results
 				}
-				if instance.LastEvalTime.After(newRule.LastEvaluation) {
-					newRule.LastEvaluation = instance.LastEvalTime
-					newGroup.LastEvaluation = instance.LastEvalTime
+
+				if alertState.LastEvaluationTime.After(newRule.LastEvaluation) {
+					newRule.LastEvaluation = alertState.LastEvaluationTime
+					newGroup.LastEvaluation = alertState.LastEvaluationTime
 				}
-				switch alert.State {
-				case "pending":
+
+				alertingRule.Duration = alertState.ProcessingTime.Seconds()
+
+				switch alertState.State {
+				case eval.Normal:
+				case eval.Pending:
 					if alertingRule.State == "inactive" {
-						alertingRule.State = "pending"
+						alertingRule.State = "firing"
 					}
-				case "firing":
+				case eval.Alerting:
 					alertingRule.State = "firing"
+				case eval.Error:
+					// handle Error case based on configuration in alertRule
```

Comment:
This and the following TODOs?

Reply:
I'm going to change the health of the alertingRule but have to think about/discuss if there's anything else we should do in these cases.